[fix][doc] Describe approximate behavior of time-based quotas (#412)
apache/pulsar-site: docs/cookbooks-retention-expiry.md

You can set a size and/or time threshold and backlog retention policy for all of the topics in a [namespace](reference-terminology.md#namespace) by specifying the namespace, a size limit and/or a time limit in second, and a policy by name.

> ::: note 

Reviewer: ```suggestion
:::note 
```

You can set a size and/or time threshold and backlog retention policy for all of the topics in a [namespace](reference-terminology.md#namespace) by specifying the namespace, a size limit and/or a time limit in second, and a policy by name.

> Note that by default, time-based backlogs are enforced periodically using an approximate method. This avoids a potentially costly scan of the backlog each time a message is produced. However, it does mean that in some cases you may observe a lack of strict enforcement. To tune this behavior you should consider using the [`backlogQuotaCheckIntervalInSeconds`][backlogquotacheckintervalinseconds] and [`preciseTimeBasedBacklogQuotaCheck`][precisetimebasedbacklogquotacheck] broker options.

Reviewer 2: ```suggestion
::: note 

By default, time-based backlogs are enforced periodically using an approximate method. This avoids a potentially costly scan of the backlog each time a message is produced. However, it does mean that in some cases you may observe a lack of strict enforcement. To tune this behavior, you should consider using the [`backlogQuotaCheckIntervalInSeconds`][backlogquotacheckintervalinseconds] and [`preciseTimeBasedBacklogQuotaCheck`][precisetimebasedbacklogquotacheck] broker options.

:::
```

Reviewer 2: This suggestions apply to other md files as well. Can you update them all?

![](/assets/retention-storage-size.svg)

If you do not have any retention period and you never have much of a backlog, the upper limit for retained messages, which are acknowledged, equals the Pulsar segment rollover period + entry log rollover period + (garbage collection interval * garbage collection ratios).

> [backlogquotacheckintervalinseconds]: https://pulsar.apache.org/reference/#/next/config/reference-configuration-broker?id=backlogquotacheckintervalinseconds
> [precisetimebasedbacklogquotacheck]: https://pulsar.apache.org/reference/#/next/config/reference-configuration-broker?id=precisetimebasedbacklogquotacheck

Reviewer 2: 1. What's the context of these two paras? Do you intend mean this?

```suggestion
To configure checks, you can use `backlogquotacheckintervalinseconds` and 
`precisetimebasedbacklogquotacheck` parameters. Details see **Configuration > Plusar > Broker** on [Pulsar Reference Site](https://pulsar.apache.org/reference).
```

2. Suggest using general link (https://pulsar.apache.org/reference) instead of specific links to reduce maintenance costs and potential errors. We've applied this strategy across all docs.

Reviewer 2: [This comment](https://github.com/apache/pulsar-site/pull/412#discussion_r1113751246) was left two weeks ago. The reason for my suggestion (as a temporary workaround) at that time was:

For example, if we want to publish a new next code release (2.12.0) with docs, the whole 2.12.x doc set is copied from `master`. If we use `https://pulsar.apache.org/reference/#/next/xxxxxxxx` in links, then the links in the old doc set will always point to the latest Reference site, which is incorrect.

 < < < < < < < < < < < < < < < <

But now you can write it as `[precisetimebasedbacklogquotacheck]: https://pulsar.apache.org/reference/#/@pulsar:version_origin@/config/reference-configuration-broker?id=precisetimebasedbacklogquotacheck` since https://github.com/apache/pulsar-site/pull/456 was merged yesterday.
